psh/touch/touch.c

```diff
@@ -41,25 +41,21 @@ int psh_touch(int argc, char **argv)
 	}
 
 	for (i = 1; i < argc; i++) {
-		file = fopen(argv[i], "r");
-		if (file == NULL) {
+		/* try to update timestamps */
+		if (utimes(argv[i], NULL) != 0) {
 			if (errno == ENOENT) {
 				/* file does not exist -> create it */
 				file = fopen(argv[i], "w");
 				if (file != NULL) {
 					fclose(file);
 					continue;
 				}
+				else {
```

Comment:
nah, don't move this code, previously it was run also if `utimes` has failed, now it's not

Reply:
also - adding strerror() to error message would be nice, but for the code to be common - we would need to switch to bare `open` instead of `fopen`